MAUI/DataGrid/Editing.md

```diff
@@ -0,0 +1,376 @@
+            RaisePropertyChanged("ShippingDate");
+        }
+    }
+
+    #endregion
+
+    #region INotifyPropertyChanged implementation
+
+    public event PropertyChangedEventHandler PropertyChanged;
+
+	private void RaisePropertyChanged (String Name)
+	{
+		if (PropertyChanged != null)
+			this.PropertyChanged (this, new PropertyChangedEventArgs (Name));
+	}
+
+    private Dictionary<string, object> storedValues;
+
+
+    public void BeginEdit()
+    {
+        this.storedValues = this.BackUp();
+    }
+
+    public void CancelEdit()
+    {
+        if (this.storedValues == null)
+            return;
+
+        foreach (var item in this.storedValues)
+        {
+            var itemProperties = this.GetType().GetTypeInfo().DeclaredProperties;
+            var pDesc = itemProperties.FirstOrDefault(p => p.Name == item.Key);
+            if (pDesc != null)
+                pDesc.SetValue(this, item.Value);
+        }
+    }
+
+    public void EndEdit()
+    {
+        if (this.storedValues != null)
+        {
+            this.storedValues.Clear();
+            this.storedValues = null;
+        }
+        Debug.WriteLine("End Edit Called");
+    }
+
+    protected Dictionary<string, object> BackUp()
+    {
+        var dictionary = new Dictionary<string, object>();
+        var itemProperties = this.GetType().GetTypeInfo().DeclaredProperties;
+        foreach (var pDescriptor in itemProperties)
+        {
+            if (pDescriptor.CanWrite)
+                dictionary.Add(pDescriptor.Name, pDescriptor.GetValue(this));
+        }
+        return dictionary;
+    }
+
+    #endregion
+}
+{% endhighlight %}
+
+
+## Editing events
+
+The SfDataGrid triggers the following events while editing:
+
+### CurrentCellBeginEdit
+
+The [SfDataGrid.CurrentCellBeginEdit]() event occurs when the CurrentCell enters into edit mode. The [DataGridCurrentCellBeginEditEventArgs]() has the following members which provides information for `SfDataGrid.CurrentCellBeginEdit` event:
+
+ * [Cancel](): When this member set to ‘true’, the event is canceled and the CurrentCell does not enter into the edit mode.
+ * [RowColumnIndex](): Gets the current row and column index of the DataGrid.
+ * [Column](): Gets the Grid Column of the SfDataGrid.
+
+To hook the `SfDataGrid.CurrentCellBeginEdit` event, follow the code example:
+
+{% highlight c# %}
+this.dataGrid.CurrentCellBeginEdit += DataGrid_CurrentCellBeginEdit;
+
+private void DataGrid_CurrentCellBeginEdit(object sender, DataGridCurrentCellBeginEditEventArgs e)
+{
+    // Editing prevented for the cell at RowColumnIndex(2,2).
+    if (e.RowColumnIndex == new Syncfusion.Maui.GridCommon.ScrollAxis.RowColumnIndex(2, 2)) 
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+### CurrentCellEndEdit
+
+The [CurrentCellEndEdit]() event occurs when the CurrentCell exits the edit mode. The [DataGridCurrentCellEndEditEventArgs]() has following members which provides information for `SfDataGrid.CurrentCellEndEdit` event:
```

Comment:
The [CurrentCellEndEdit]() event occurs when the CurrentCell exits the edit mode. The [DataGridCurrentCellEndEditEventArgs]() has following members that provides information for `SfDataGrid.CurrentCellEndEdit` event:

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/6e1845d0-6551-4b46-8030-7f9a5d71e86b)